libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -455,48 +463,48 @@ method unsubscribe*(g: GossipSub,
 
 method publish*(g: GossipSub,
                 topic: string,
-                data: seq[byte]) {.async.} =
-  await procCall PubSub(g).publish(topic, data)
+                data: seq[byte]): Future[int] {.async.} =
+  # base returns always 0
+  discard await procCall PubSub(g).publish(topic, data)
   trace "about to publish message on topic", name = topic,
                                              data = data.shortLog
 
   var peers: HashSet[string]
-  # TODO: we probably don't need to try multiple times
-  if data.len > 0 and topic.len > 0:
-    for _ in 0..<5: # try to get peers up to 5 times
-      if peers.len > 0:
-        break
-
-      if topic in g.topics: # if we're subscribed to the topic attempt to build a mesh
-        await g.rebalanceMesh(topic)
-        peers = g.mesh.getOrDefault(topic)
-      else: # send to fanout peers
-        await g.replenishFanout(topic)
-        if topic in g.fanout:
-          peers = g.fanout.getOrDefault(topic)
-          # set the fanout expiry time
-          g.lastFanoutPubSub[topic] = Moment.fromNow(GossipSubFanoutTTL)
 
-      # wait a second between tries
-      await sleepAsync(1.seconds)
+  if topic.len > 0: # data could be 0/empty
```

Comment:
we shouldn't publish empty data, at least I can't think of any use for it and I haven't seen this in other implementations.

Reply:
why not tho? a signal of some sort?

Reply:
go has no filtering by size at all btw:
https://github.com/libp2p/go-libp2p-pubsub/blob/8945f91465bc3ba418688684a2c29c384cd79647/topic.go#L160

Reply:
We can, my only concern is how would other clients handle it, since it's not spelled out in the spec, this is up for interpretation and some will handle it others won't?

Reply:
Rust also does not discriminate:
```
 /// Content of the message. Its meaning is out of scope of this library.
    pub data: Vec<u8>,
```

Reply:
from https://github.com/libp2p/rust-libp2p/blob/823c74790434213e00e7adf15992a2f43cf0b52b/protocols/gossipsub/src/protocol.rs#L345